content/blog/tidymodels-2023-q1/index.Rmd

```diff
@@ -0,0 +1,203 @@
+#> ℹ In argument: `.estimate = metric_fn(truth = vs, estimate = am, na_rm =
+#>   na_rm)`.
+#> Caused by error in `validate_class()`:
+#> ! `truth` should be a factor but a numeric was supplied.
+```
+
+which now errors much better
+
+```{r}
+#| error: TRUE
+mtcars |>
+  accuracy(vs, am)
+```
+
+Lastly, one of the biggest improvements came in recipes, which now shows which step caused the error instead of saying it happened in `prep()` or `bake()`. This is a huge improvement since you are likely to have many steps.
+
+Before
+
+``` r
+recipe(~., data = ames) |>
+  step_novel(Neighborhood, new_level = "Gilbert") |>
+  prep()
+#> Error in `prep()`:
+#> ! Columns already contain the new level: Neighborhood
+```
+
+Now
+
+```{r}
+#| error: TRUE
+recipe(~., data = ames) |>
+  step_novel(Neighborhood, new_level = "Gilbert") |>
+  prep()
+```
+
+While these changes might seem trivial, they make a big difference. Especially when the errors are coming when you are using `fit_resamples()` or `tune_grid()`.
+
+## Things are getting faster 
+
+As we have written about in [The tidymodels is getting a whole lot faster](https://www.simonpcouch.com/blog/speedups-2023/) and [Writing performant code with tidy tools
+](https://www.tidyverse.org/blog/2023/04/performant-packages/), we have been working on tightening up the performance of the tidymodels code. These changes are mostly related to the infrastructure code, meaning that the speedup will bring you to closer underlying implementations.
+
+A different kind of speedup is found with the addition of the [step_pca_truncated()](https://embed.tidymodels.org/reference/step_pca_truncated.html) step added in the embed package.
+
+[Principal Component Analysis](https://en.wikipedia.org/wiki/Principal_component_analysis) is a really powerful and fast method for dimensionality reduction of large data sets. However, for data with many columns, it can be computationally expensive to calculate all the principal components. `step_pca_truncated()` works in much the same way as `step_pca()` but it only calculates the number of components it needs
+
+```{r}
+pca_normal <- recipe(Sale_Price ~ ., data = ames) |>
+  step_dummy(all_nominal_predictors()) |>
+  step_pca(all_numeric_predictors(), num_comp = 3)
+
+pca_truncated <- recipe(Sale_Price ~ ., data = ames) |>
+  step_dummy(all_nominal_predictors()) |>
+  step_pca_truncated(all_numeric_predictors(), num_comp = 3)
+```
+
+```{r}
+tictoc::tic()
+prep(pca_normal) |> bake(ames)
+tictoc::toc()
+```
+
+```{r}
+tictoc::tic()
+prep(pca_truncated) |> bake(ames)
+tictoc::toc()
+```
+
+The speedup will be orders of magnitude larger for very wide data.
+
+## Acknowledgements
+
+We'd like to thank those in the community that contributed to tidymodels in the last quarter:
+
+```{r}
+#| echo: FALSE
+#| results: asis
+
+cat(tm_data$thanks, sep = "\n")
+```
+
+We're grateful for all of the tidymodels community, from observers to users to contributors, and wish you all a happy new year!
```

Comment:
```suggestion
We're grateful for all of the tidymodels community, from observers to users to contributors. Happy modeling!
```

I belieeeeve the "happy new year" is an artifact from last quarter's post?

Reply:
it sure is! 😄